python/cudf/cudf/utils/applyutils.py

```diff
@@ -104,10 +127,15 @@ def run(self, df, **launch_params):
         bound = self.sig.bind(**args)
         # Launch kernel
         self.launch_kernel(df, bound.args, **launch_params)
+        # Prepare pessimistic nullmask
+        out_mask = make_aggregate_nullmask(df)
         # Prepare output frame
         outdf = df.copy()
         for k in sorted(self.outcols):
             outdf[k] = outputs[k]
+            if out_mask is not None:
+                outdf[k] = outdf[k].set_mask(out_mask.data.mem)
```

Comment:
This actually wants a Buffer instead of the underlying array (though it's handled):
```suggestion
                outdf[k] = outdf[k].set_mask(out_mask.data)
```